File: mega_sena.c
#include <stdio.h>

void ordenar(int *nums, int tam){
    for(int i = 1; i < tam; i++){
        int temp = nums[i];
        int j = i - 1;

        while(j >= 0 && nums[j] > temp){
            nums[j + 1] = nums[j];
            j--;
        }
        nums[j + 1] = temp;
    }
}

int main(){
    int nums[36];
    int tam = 0;
    int num;

    while(scanf("%d", &num) == 1){
        int repetido = 0;

        for(int i = 0; i < tam; i++){
            if(nums[i] == num){
                repetido = 1;
                break;
            }
        }
        if(!repetido) nums[tam++] = num;
    }

    ordenar(nums, tam);

    for(int i = 0; i < tam; i++){
        printf("%d\n", nums[i]);
    }

    printf("Total de jogos completos: %d\n", tam / 6);

    return 0;
}

File: mega_sena.py
nums = list(map(int, input().split()))
unicos = sorted(set(nums))
for n in unicos:
    print(n)
print(f"Total de jogos completos: {len(unicos) // 6}")

# Divide a entrada pelos espaços
#.split()

# Converte cada parte para inteiro
#map(int, ...)

# Transforma o resultado em lista
#list(...)

# Remove repetidos
#set(...)

# Ordena os números
#sorted(...)
